torch/csrc/dynamo/eval_frame.c

```diff
@@ -293,8 +294,34 @@ static PyObject* dynamo_call_callback(
   frame->locals = (PyObject*)framelocals_mapping_to_dict(locals);
 
   PyObject* cache_entry_pyobj = CacheEntry_to_obj(cache_entry);
+
+  // preserve python random state
+  PyObject* prev_rng_state = random_getstate(random_module());
+
   PyObject* res = PyObject_CallFunction(
       callable, "OOO", frame, cache_entry_pyobj, frame_state);
+
+  // store error if there was one
+#if IS_PYTHON_3_12_PLUS
+  PyObject* exc = PyErr_GetRaisedException();
+#else
+  PyObject *exc_type, *exc_value, *exc_traceback;
```

Comment:
Are these objects owned an needing recounting?

Reply:
No need, PyErr_SetRaisedException/PyErr_Restore steals the reference.